### PR TITLE
Implemented extra boolean param for rollback()

### PR DESCRIPTION
**Pull request by edevil**
Extra parameter allows all migrations to be rolled back.
Addresses https://github.com/tgriesser/knex/issues/2844

### CODE REVIEW

File: test/integration/migrate/index.js [hunk 1]
```diff
@@ -331,6 +331,55 @@ module.exports = function(knex) {
...
+              directory: [
+                'test/integration/migrate/test',
+                'test/integration/migrate/test2',
+              ],
+            });
+          });
+      });
+
+      it('should delete all batches from the migration log', function() {
+        debugger;
```
Comment (kibertoad) [inline test/integration/migrate/index.js]: probably this shouldn't be committed :)

File: test/integration/migrate/index.js [hunk 2]
```diff
@@ -331,6 +331,55 @@ module.exports = function(knex) {
...
+            return knex.migrate.latest({
+              directory: [
+                'test/integration/migrate/test',
+                'test/integration/migrate/test2',
+              ],
+            });
+          });
+      });
+
+      it('should delete all batches from the migration log', function() {
```
Comment (kibertoad) [inline test/integration/migrate/index.js]: why not use arrow function?

File: test/integration/migrate/index.js [hunk 3]
```diff
@@ -331,6 +331,55 @@ module.exports = function(knex) {
...
+            {
+              directory: [
+                'test/integration/migrate/test',
+                'test/integration/migrate/test2',
+              ],
+            },
+            true
+          )
+          .spread(function(batchNo, log) {
+            console.log(`Batch ${batchNo}, log: ${JSON.stringify(log)}`);
```
Comment (kibertoad) [inline test/integration/migrate/index.js]: is this needed?

File: types/knex.d.ts [hunk 4]
```diff
@@ -829,7 +829,7 @@ declare namespace Knex {
   interface Migrator {
     make(name: string, config?: MigratorConfig): Bluebird<string>;
     latest(config?: MigratorConfig): Bluebird<any>;
-    rollback(config?: MigratorConfig): Bluebird<any>;
+    rollback(config?: MigratorConfig, all?: boolean): Bluebird<any>;
```
Comment (kibertoad) [inline types/knex.d.ts]: haha, nice touch!

**Comment by kibertoad** [comment]
Could you also create a PR in documentation repo?

File: test/integration/migrate/index.js (hunk 3 above)
Comment (edevil) [reply test/integration/migrate/index.js]: Oops, sorry.

File: test/integration/migrate/index.js (hunk 1 above)
Comment (edevil) [reply test/integration/migrate/index.js]: Crap, sorry...

**Comment by kibertoad** [comment]
LGTM, we can merge it after documentation is created.

**Comment by edevil** [comment]
@kibertoad Great! The docs PR is at https://github.com/knex/documentation/pull/170

**Comment by kibertoad** [comment]
Thank you a lot!

File: test/integration/migrate/index.js (hunk 2 above)
Comment (ricardograca) [reply test/integration/migrate/index.js]: Because of [this](https://mochajs.org/#arrow-functions)?

Comment (kibertoad) [reply test/integration/migrate/index.js]: well yeah when it's needed, but when it's not?..

**Pull request merged**